lib/pg_ha_migrations/safe_statements.rb

```diff
@@ -286,6 +299,96 @@ def safe_create_partitioned_table(table, key:, type:, infer_primary_key: nil, **
     end
   end
 
+  def safe_partman_create_parent(table, key:, interval:, infinite_time_partitions: true, inherit_privileges: true, **options)
+    raise ArgumentError, "Expected <key> to be present" unless key.present?
+    raise ArgumentError, "Expected <interval> to be present" unless interval.present?
+
+    invalid_options = options.keys - PARTMAN_CREATE_PARENT_OPTIONS - PARTMAN_UPDATE_CONFIG_OPTIONS
+
+    raise ArgumentError, "Unrecognized optional argument(s): #{invalid_options}" unless invalid_options.empty?
+
+    if ActiveRecord::Base.connection.postgresql_version < 11_00_00
+      raise PgHaMigrations::InvalidMigrationError, "Native partitioning with partman not supported on Postgres databases before version 11"
+    end
+
+    fully_qualified_table_name = _fully_qualified_name(table)
+
+    create_parent_options = options.slice(*PARTMAN_CREATE_PARENT_OPTIONS)
+    update_config_options = options.slice(*PARTMAN_UPDATE_CONFIG_OPTIONS)
+
+    if create_parent_options[:template_table].present?
+      create_parent_options[:template_table] = _fully_qualified_name(create_parent_options[:template_table])
+    end
+
+    create_parent_options = create_parent_options.merge(
+      parent_table: fully_qualified_table_name,
+      control: key,
+      type: "native",
+      interval: interval,
+    ).compact
+
+    update_config_options = update_config_options.merge(
+      infinite_time_partitions: infinite_time_partitions,
+      inherit_privileges: inherit_privileges,
+    ).compact
+
+    create_parent_sql = create_parent_options.map { |k, v| "p_#{k} := #{quote(v)}" }.join(", ")
+
+    connection.execute("SELECT #{_quoted_partman_schema}.create_parent(#{create_parent_sql})")
+
+    safe_partman_update_config(fully_qualified_table_name, **update_config_options)
+  end
+
+  def safe_partman_update_config(table, **options)
+    invalid_options = options.keys - PARTMAN_UPDATE_CONFIG_OPTIONS
+
+    raise ArgumentError, "Unrecognized argument(s): #{invalid_options}" unless invalid_options.empty?
+
+    PgHaMigrations::PartmanConfig.schema = _quoted_partman_schema
```

Comment:
This feels pretty weird IMO.

Reply:
I agree, but I do think it's a pretty nice user experience to auto-discover the schema. The alternative here would maybe be to define a config value for the partman schema that defaults to `partman` but can be overridden:

```ruby
PgHaMigrations.configure do |config|
  config.partman_schema = "public"
end
```

Reply:
Yeah, that is kinda convenient.